editor/components/document-outline/index.js

```diff
@@ -57,10 +57,26 @@ const getHeadingLevel = heading => {
 	}
 };
 
+const computeOutlineHeadings = ( blocks = [], path = [] ) => {
+	return flatMap( blocks, ( block = {} ) => {
+		if ( block.name === 'core/heading' ) {
+			return {
+				...block,
+				path,
+				level: getHeadingLevel( block ),
+				isEmpty: isEmptyHeading( block ),
+			};
+		}
+		return isEmpty( block.innerBlocks ) ?
```

Comment:
Is the `isEmpty` check necessary? `flatMap` on an empty array should return an empty array anyways.